tests/test_astronomer_commander.py

```diff
@@ -0,0 +1,61 @@
+from tests.helm_template_generator import render_chart
+import pytest
+from . import supported_k8s_versions
+import jmespath
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+def test_astronomer_commander_deployment(kube_version):
+    """Test that helm renders a good deployment template for astronomer/commander."""
+    docs = render_chart(
+        kube_version=kube_version,
+        show_only=["charts/astronomer/templates/commander/commander-deployment.yaml"],
+    )
+
+    assert len(docs) == 1
+    doc = docs[0]
+    assert doc["kind"] == "Deployment"
+    assert doc["apiVersion"] == "apps/v1"
+    assert doc["metadata"]["name"] == "RELEASE-NAME-commander"
+
+    envVarNames = jmespath.search("spec.template.spec.containers[0].env[*].name", doc)
+    assert "COMMANDER_UPGRADE_TIMEOUT" in envVarNames
+    upgradeTimeoutIndex = envVarNames.index("COMMANDER_UPGRADE_TIMEOUT")
+    assert (
+        doc["spec"]["template"]["spec"]["containers"][0]["env"][upgradeTimeoutIndex][
+            "value"
+        ]
+        == "300"
+    )
```

Comment:
Do you think this is easier to read if we reshape the `env` list using a dict comprehension?
```suggestion
    assert len(doc["spec"]["template"]["spec"]["containers"]) == 1
    env_vars = {x['name']: x['value'] for x in doc["spec"]["template"]["spec"]["containers"][0]['env']}
    assert env_vars['COMMANDER_UPGRADE_TIMEOUT'] == "300"
```

Reply:
Also I added a length assertion here because we are making assumptions about what `[0]` is.